lib/ansible/modules/cloud/docker/docker_volume.py

```diff
@@ -0,0 +1,245 @@
+options:
+  name:
+    description:
+      - Name of the volume to operate on.
+    required: true
+    aliases:
+      - volume_name
+
+  driver:
+    description:
+      - Specify the type of volume. Docker provides the C(local) driver, but 3rd party drivers can also be used.
+    default: local
+
+  driver_options:
+    description:
+      - >
+        Dictionary of volume settings. Consult docker docs for valid options and values:
+        U(https://docs.docker.com/engine/reference/commandline/volume_create/#driver-specific-options)
+
+  force:
+    description:
+      - With state C(present) causes the volume to be deleted and recreated if the volume already
+        exist and the driver, driver options or labels differ.
+        This will cause any data in the existing volume to be lost.
+    type: bool
+    default: 'no'
+
+  state:
+    description:
+      - C(absent) deletes the volume.
+      - C(present) creates the volume, if it does not already exist.
+    default: present
+    choices:
+      - absent
+      - present
+
+extends_documentation_fragment:
+    - docker
+
+author:
+    - Alex Grönholm (@agronholm)
+
+requirements:
+    - "python >= 2.6"
+    - "docker-py >= 1.10.0"
+    - "The docker server >= 1.9.0"
+'''
+
+EXAMPLES = '''
+- name: Create a volume
+  docker_volume:
+    name: volume_one
+
+- name: Remove a volume
+  docker_volume:
+    name: volume_one
+    state: absent
+
+- name: Create a volume with options
+  docker_volume:
+    name: volume_two
+    driver_options:
+      type: btrfs
+      device: /dev/sda2
+'''
+
+RETURN = '''
+facts:
+    description: Volume inspection results for the affected volume.
+    returned: success
+    type: dict
+    sample: {}
+'''
+
+from ansible.module_utils.six import iteritems
+from ansible.module_utils.docker_common import DockerBaseClass, AnsibleDockerClient
+
+
+class TaskParameters(DockerBaseClass):
+    def __init__(self, client):
+        super(TaskParameters, self).__init__()
+        self.client = client
+
+        self.volume_name = None
+        self.driver = None
+        self.driver_options = None
+        self.labels = None
+        self.force = None
+        self.debug = None
+
+        for key, value in iteritems(client.module.params):
+            setattr(self, key, value)
+
+
+class DockerVolumeManager(object):
+
+    def __init__(self, client):
+        self.client = client
+        self.parameters = TaskParameters(client)
+        self.check_mode = self.client.check_mode
+        self.results = {
+            u'changed': False,
+            u'actions': []
+        }
+        self.diff = self.client.module._diff
+
+        self.existing_volume = self.get_existing_volume()
+
+        state = self.parameters.state
+        if state == 'present':
+            self.present()
+        elif state == 'absent':
+            self.absent()
+
+    def get_existing_volume(self):
+        volumes = self.client.volumes()
```

Comment:
Wrap calls to self.client.X in try/except block. On APIError call self.client.fail() with an error message and exception details.